doc/source/notebooks/advanced/mcmc.pct.py

```diff
@@ -589,3 +598,94 @@ def run_chain_fn():
     ax.hist(parameter_samples[name_to_index[hyp]], bins=20)
     ax.set_title(hyp)
 plt.tight_layout()
+
+# %% [markdown]
+# ## Prior on constraint and unconstrained parameters
+
+# %% [markdown]
+# GPflow's :class:`Parameter` class provides options for setting a prior. :class:`Parameter` wraps a constrained tensors and
+# provides computation of the gradient with respect to unconstrained transformation of that tensor.
+# The user can set a prior either in **constrained** space or **unconstrained** space.
+
+# %% [markdown]
+# By default prior for the `Parameter` is set for _constrained_ space. To set a prior do the following:
+
+# %%
+prior_distribution = tfd.Normal(f64(0.0), f64(1.0))
+_ = gpflow.Parameter(1.0, prior_on="unconstrained", prior=prior_distribution)
+_ = gpflow.Parameter(1.0, prior_on="constrained", prior=prior_distribution)
+
+# %% [markdown]
+# When a prior is set in *constrained* space, the `gpflow.optimizers.SamplingHelper` compensates the prior density with a Jacobian of the transformation used by the parameter. For priors in the *unconstrained* space the `gpflow.optimizers.SamplingHelper` will not apply adjustments to the prior density.
+
+# %% [markdown]
+# Below we repeat the same experiment as before, but with some priors defined in the `unconstrained` space. Since we are using an "exp" transform, a Log-normal prior on a constrained parameter corresponds to a Normal prior on the unconstrained space:
+
+# %%
+gpflow.config.set_default_positive_bijector("exp")
+gpflow.config.set_default_positive_minimum(1e-6)
+
+rng = np.random.RandomState(42)
+data = synthetic_data(30, rng)
+
+kernel = gpflow.kernels.Matern52(lengthscales=0.3)
+meanf = gpflow.mean_functions.Linear(1.0, 0.0)
+model = gpflow.models.GPR(data, kernel, meanf)
+model.likelihood.variance.assign(0.01)
+
+mu = f64(0.0)
+std = f64(4.0)
+one = f64(1.0)
+
+model.kernel.lengthscales.prior_on = "unconstrained"
+default_prior = tfd.Normal(mu, std)
+model.kernel.lengthscales.prior = default_prior
```

Comment:
```suggestion
model.kernel.lengthscales.prior = tfd.Normal(mu, std)
```
use it everywhere or nowhere?:)

Reply:
That was a Fergus change ;)

Reply:
Hah true, my intention was to use it everywhere (unless there is a downside to re-using the same instance of tfd.Normal, but I think it's fine in this context)